src/lsdb/loaders/dataframe/read_dataframe.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from hipscat.catalog import CatalogType
+
+from lsdb.catalog.dataset.dataset import Dataset
+from lsdb.loaders.dataframe.dataframe_catalog_loader import DataframeCatalogLoader
+from lsdb.loaders.hipscat.hipscat_loader_factory import CatalogTypeVar
+
+
+def read_dataframe(
+    path: str,
+    catalog_name: str = "",
+    catalog_type: CatalogType | None = None,
+    ra_column: str = "ra",
+    dec_column: str = "dec",
+    threshold: int = 50,
+) -> CatalogTypeVar | Dataset:
```

Comment:
Woo typing. I'm guessing the return type here is copied from `read_hipscat` but that's a bit different. It can read any type of catalog, so it could return an `AssociationCatalog` for example. (It can't now, but that's because it's not finished.) But this just returns a `Catalog` so that should be the return type.